src/ArborX_DistributedSearchTree.hpp

```diff
@@ -152,6 +155,28 @@ DistributedSearchTree<DeviceType>::DistributedSearchTree(
   _top_tree_size = accumulate(space, _bottom_tree_sizes, 0);
 }
 
+template <typename DeviceType>
+class DistributedSearchTree<
+    DeviceType, std::enable_if_t<Details::is_device_type<DeviceType>::value>>
+    : public DistributedSearchTree<typename DeviceType::memory_space>
+{
+public:
+  using device_type = DeviceType;
+  DistributedSearchTree() = default;
```

Comment:
Is there a reason to allow a default constructed `DistributedSearchTree` now?
The base class doesn't have a default constructor so using it should even not work at all (I tried to be sure).

Reply:
I didn't realize I had not provided a default constructor.  In that case it is deleted anyway.  I can remove if you like.

Reply:
Yes, I think it is reasonable to expect that explicitly defaulted constructors are usable and I don't see why a default constructed object would make sense here.

Reply:
I removed.  I am not sure if you understand it was without effect.

Reply:
It's much clearer to me that the object can't be default-constructed.